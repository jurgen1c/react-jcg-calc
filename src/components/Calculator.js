import { useState } from 'react';
import calculate from '../logic/calculate';

import Display from './Display';
import ButtonPanel from './ButtonPanel';

const Calculator = () => {
  const [state, setState] = useState({ total: '', next: '', operation: '' });

  const handleClick = btnName => {
    const result = calculate(state, btnName);
    setState(result);
  };

  return (
    <div className="container">
      <div>
        <h2>Lets do some Math!</h2>
      </div>
      <div className="app">
        <Display
          result={String(state.total)}
          next={String(state.next)}
          operation={state.operation}
        />
        <ButtonPanel clickHandler={handleClick} />
      </div>
    </div>
  );
};

export default Calculator;
