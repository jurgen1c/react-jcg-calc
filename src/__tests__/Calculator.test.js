import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

test('Componenet renders correctly', () => {
  const component = renderer.create(
    <Calculator />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
