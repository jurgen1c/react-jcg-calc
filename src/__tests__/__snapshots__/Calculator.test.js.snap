// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componenet renders correctly 1`] = `
<div
  className="container"
>
  <div>
    <h2>
      Lets do some Math!
    </h2>
  </div>
  <div
    className="app"
  >
    <div
      className="display"
    >
      
      
      
    </div>
    <div
      className="buttonPanel"
    >
      <div>
        <button
          className="sc-bdfBwQ iFFVnC btn"
          name="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="sc-gsTCUz ivWJYG btn"
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="sc-dlfnbm bOOCLx btn"
          name="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="sc-hKgILt ATUro btn"
          name="รท"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div>
        <button
          className="sc-eCssSg cwRZbc btn"
          name="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="sc-jSgupP cNYwkx btn"
          name="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="sc-gKsewC huJzkI btn"
          name="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="sc-iBPRYJ kKXfUf btn"
          name="X"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
      <div>
        <button
          className="sc-fubCfw iLRLpB btn"
          name="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="sc-pFZIQ hvhYqa btn"
          name="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="sc-jrAGrp bFqaqO btn"
          name="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="sc-kEjbxe bnLhHi btn"
          name="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div>
        <button
          className="sc-iqHYGH NkWAk btn"
          name="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="sc-crrsfI fJMhWB btn"
          name="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="sc-dQppl eJqEGn btn"
          name="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="sc-bqyKva daHCJL btn"
          name="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div>
        <button
          className="sc-kstrdz fExTGg o-btn"
          name="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="sc-hBEYos EjdWo btn"
          name="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="sc-fodVxV fnuKlM btn"
          name="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
